src/components/FileMachineProvider.tsx

```diff
@@ -306,6 +308,18 @@ export const FileMachineProvider = ({
               },
             })
           }
+
+          // Either way, we want to overwrite the defaultUnit project setting
+          // with the sample's setting.
+          if (data.sampleUnits) {
+            settings.send({
+              type: 'set.modeling.defaultUnit',
+              data: {
+                level: 'project',
+                value: data.sampleUnits,
+              },
+            })
+          }
```

Comment:
What happens when a user goes back to a file they own outside of the sample? Will the units revert?